aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -113,21 +110,21 @@ NestedTensorImpl::NestedTensorImpl(
     at::Tensor buffer,
     at::Tensor nested_size_tensor,
     at::Tensor nested_stride_tensor,
-    const std::vector<int64_t>& offsets)
+    std::vector<int64_t> offsets)
     : TensorImpl(
+          Storage(buffer.storage()),
           generate_nested_key_set(buffer),
-          buffer.dtype(),
-          buffer.device()),
-      buffer_(std::move(buffer)),
+          buffer.dtype()),
+      buffer_size_(buffer.unsafeGetTensorImpl()->sizes().vec()),
```

Comment:
If this is always one element, why not just store it as a int64 directly? That would be faster and simplify your like for convertion between ArrayRef/vector.

Reply:
Yeah Joel mentioned this and just updated

Reply:
Ho ok, I guess the change is not pushed yet in the PR